tests/prague/eip7692_eof_v1/eip7069_extcall/test_address_space_extension.py

```diff
@@ -0,0 +1,226 @@
+slot_target_returndata = next(_slot)
+
+value_legacy_success = 1
+value_eof_success = 0
+value_eof_fail = 1
+
+
+@pytest.mark.parametrize(
+    "target_address",
+    (
+        pytest.param(b"", id="zero"),
+        pytest.param(b"\xc0\xde", id="short"),
+        pytest.param(b"\x78" * 20, id="mid_20"),
+        pytest.param(b"\xff" * 20, id="max_20"),
+        pytest.param(b"\x01" + (b"\x00" * 20), id="min_ase"),
+        pytest.param(b"\x5a" * 28, id="mid_ase"),
+        pytest.param(b"\x5a" * 32, id="full_ase"),
+        pytest.param(b"\xff" * 32, id="max_ase"),
+    ),
+)
+@pytest.mark.parametrize(
+    "target_account_type", ("empty", "EOA", "LegacyContract", "EOFContract"), ids=lambda x: x
+)
+@pytest.mark.parametrize(
+    "target_opcode",
+    (
+        Op.CALL,
+        Op.CALLCODE,
+        Op.STATICCALL,
+        Op.DELEGATECALL,
+        Op.EXTCALL,
+        Op.EXTDELEGATECALL,
+        Op.EXTSTATICCALL,
+    ),
+)
+def test_address_space_extension(
+    state_test: StateTestFiller,
+    target_address: bytes,
+    target_opcode: Op,
+    target_account_type: str,
+):
+    """
+    Test contacts with possibly extended address and fail if address is too large
+    """
+    env = Environment()
+
+    ase_address = len(target_address) > 20
+    stripped_address = target_address[-20:] if ase_address else target_address
+    if ase_address and target_address[0] == b"00":
+        raise ValueError("Test instrumentation requires target addresses trim leading zeros")
+
+    match target_opcode:
+        case Op.CALL | Op.CALLCODE:
+            call_suffix = [0, 0, 0, 0, 0]
+            ase_ready_opcode = False
+        case Op.DELEGATECALL | Op.STATICCALL:
+            call_suffix = [0, 0, 0, 0]
+            ase_ready_opcode = False
+        case Op.EXTCALL:
+            call_suffix = [0, 0, 0]
+            ase_ready_opcode = True
+        case Op.EXTDELEGATECALL | Op.EXTSTATICCALL:
+            call_suffix = [0, 0]
+            ase_ready_opcode = True
+        case _:
+            raise ValueError("Unexpected opcode ", target_opcode)
+
+    pre = {
+        TestAddress: Account(
+            balance=1000000000000000000000,
+            nonce=1,
+        ),
+        address_caller: Account(
+            code=(
+                Op.MSTORE(0, Op.PUSH32(target_address))
+                + Op.SSTORE(
+                    slot_top_level_call_status,
+                    Op.CALL(50000, address_target, 0, 0, 32, 0, 0),
+                )
+                + Op.STOP()
+            ),
+            nonce=1,
+        ),
+        address_target: Account(
+            code=Container(
+                sections=[
+                    Section.Code(
+                        code=Op.SSTORE(
+                            slot_target_call_status,
+                            target_opcode(Op.CALLDATALOAD(0), *call_suffix),
+                        )
+                        + Op.SSTORE(slot_target_returndata, Op.RETURNDATALOAD(0))
```

Comment:
Same thing here, emvone is exceptionally aborting execution when reaching `RETURNDATALOAD`.
Changing to RETURNDATACOPY + RETURNDATASIZE + MLOAD solves the issue here too.

Reply:
See prior comment.  evmone is out of spect at the moment.

Reply:
This is also an emergent behavior of EXTDELEGATECALL failing to call a legacy contract, and thus zero returndata is in the return data buffer, hence all calls to RETURNDATALOAD should return zeros.